sql/driver/driver.go

```diff
@@ -52,5 +63,27 @@ func (d *roachDriver) Open(dsn string) (driver.Conn, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &conn{sender: sender}, nil
+	conn := conn{sender: sender}
+	if err := d.applySettings(&conn); err != nil {
+		return nil, err
+	}
+	return &conn, nil
+}
+
+func (d *RoachDriver) applySettings(c *conn) error {
+	if d.Settings.DB != "" {
+		if _, err := c.Exec("SET DATABASE = "+d.Settings.DB, []driver.Value{}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// SetSessionSettings for the DB driver. Since the same driver
+// is shared across multiple DBs this will affect all DBs sharing
+// the same driver.
+func SetSessionSettings(db *sql.DB, settings SessionSettings) {
```

Comment:
I think exposing setting session settings like this isn't the right pattern. For one, this won't apply to any existing connections. And it differs from the way other go sql drivers handle this. The more standard way is to specify global settings in the `dsn` string passed to `Open`. We can interpret that string however we want and we're currently interpreting it as a URL. Adding a `database=<foo>` parameter looks easy.